aot/src/main/java/com/dylibso/chicory/experimental/aot/AotCompiler.java

```diff
@@ -388,12 +392,23 @@ private byte[] compileMachineCallClass() {
                 });
 
         // static implementation for Machine.call()
-        emitFunction(
-                classWriter,
-                "call",
-                MACHINE_CALL_METHOD_TYPE,
-                true,
-                this::compileMachineCallInvoke);
+        Consumer<InstructionAdapter> callMethod;
+        if (functionTypes.size() < MAX_MACHINE_CALL_METHODS) {
+            callMethod = asm -> compileMachineCallInvoke(asm, 0, functionTypes.size());
+        } else {
```

Comment:
ok, so here we are saying that after a fixed number of functions we can dispatch the call, I assume that this is the building block to split.

Reply:
Correct. We generate an outer dispatch method that calls the correct inner dispatch method.